fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/leaderboard/TimeWindowLeaderboardIndexMaintainer.java

```diff
@@ -155,12 +157,18 @@ protected void saveSubDirectory(@Nonnull TimeWindowLeaderboardSubDirectory subdi
         state.transaction.set(extraSubspace.pack(SUB_DIRECTORY_PREFIX.addAll(subdirectory.getGroup())), subdirectory.toProto().toByteArray());
     }
 
+    @Override
+    public RecordCursor<IndexEntry> scan(@Nonnull IndexScanType scanType, @Nonnull TupleRange range,
+                                         @Nullable byte[] continuation, @Nonnull ScanProperties scanProperties) {
+        return scan(new IndexScanRange(scanType, range), continuation, scanProperties);
```

Comment:
Should this extract the leaderboard type and timestamp from `range` if this is called with a `BY_TIME_WINDOW` scan type (and then pass a `TimeWindowScanRange` instead of an `IndexScanRange`)? Otherwise, I think this will break callers who call `scan` on time window leaderboard indexes directly (instead of, if there is one, some higher level abstraction, like a query plan).

Reply:
I guess that is best for a little while, at least, for compatibility. I added back the old code under conditions for that case.